Issue #2971: VisibilityModifier allowPublicFinalFields

som-snytt opened the pull request:
#2971

Loosens the restriction for `allowPublicImmutableFields` that the
enclosing class must be final.

For `allowPublicFinalFields`, the field must be final and immutable.

The new property also defaults to true.

romani commented:
commit should be single lined, please change commit message.

som-snytt commented:
Thanks, sorry I didn't notice the developer page on the sourceforge site.

romani commented:
1) please fix CIs

2) please update xdoc files documentation, with config+code examples for new option. As options are very similar to each other, we need to help user to see a difference by examples.

3) Please generate html report with your option on opensource projects to make sure there are no exceptions and reported violations as ok.
Please use this tool - https://github.com/checkstyle/contribution/tree/master/checkstyle-tester , please read README file.

4) Please generate diff-report of execution without your option and with you options to make sure there are no regression.
Please use this tool - https://github.com/checkstyle/contribution/tree/master/patch-diff-report-tool

romani commented:
@som-snytt , do you need help ?

FYI: next release will happen at 25th March.

som-snytt commented:
@romani Thanks for the ping. I hadn't seen your comment. I'll try to make time. Nrs 3 & 4 sound like time-consuming; I'll cry for help if necessary.

romani commented:
> Nrs 3 & 4 sound like time-consuming

NO, all you need is just do clone of such repos , change config and launch them.
Let me know if you have problem with them or do not understand smth.

romani commented:
@som-snytt , ping .

som-snytt commented:
Thanks, just a busy time at the day job. Sorry I missed the last release. Still on my radar, or maybe sonar.

romani commented:
@som-snytt , To resolve problem on Travis , please rebase your changes on our latest master.

romani commented:
@som-snytt , ping.

romani commented:
@som-snytt , next release in 7 days, lets make it.
You just need to fix documentation and  launch few shell scripts.

romani commented:
@som-snytt , if you do not want to finish this PR please let me know. Other contributors could do this.
Right now you block issue resolving. If I have no reply from you in next month - I will reject your PR.

som-snytt commented:
@romani I'll spare a couple of hours tonight. I see the failure includes cobertura after the rebase. That's too bad, because it was non-trivial to get coverage to 100%. Sorry I don't have much FOSS budget to donate. If someone is ready to push this over the goal line, please comment and I'll close.

som-snytt commented:
Edit: actually, it looks like I rebased two weeks ago from the wrong machine. Such are the perils of `push -f`. I'll re-rebase at home later. I'd fixed cobertura and added docs.

som-snytt commented:
Running checkstyle-tester on just guava against this branch and master results in a diff with a rule name check. Is that expected? (Out of time for now.)
```
< <error line="154" severity="warning" message="Javadoc comment at column 46 has parse error. Missed HTML close tag &apos;V&apos;. Sometimes it means that close tag missed for one of previous tags." source="com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTagContinuationIndentationCheck"/>
---
> <error line="154" severity="warning" message="Javadoc comment at column 46 has parse error. Missed HTML close tag &apos;V&apos;. Sometimes it means that close tag missed for one of previous tags." source="com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck"/>
```

romani commented:
Strange that diff is in Checks that you did not touch.
Please make it sure that you run a tool with config only for Check that you change. No other Checks should be activated.
If still a problem pleaese share with me details of what and how you launch.

rnveach commented:
@romani I don't know if those are the exact checks since the PR doesn't have that information, but you and me have seen this issue in atleast one other PR where the checks seemed to be run in a different order. See https://github.com/checkstyle/checkstyle/pull/2864#issuecomment-179580723

som-snytt commented:
I didn't configure for only this Visibility check. I'll update later.

romani commented:
@som-snytt , please let us know if you need help or some explanations on how to setup checkstyle-tester project.

romani commented:
looks like original author do not have time for this PR, assigned to @MEZk .

som-snytt commented:
There's a terminal on my desktop at work still open to checkstyle, but please, yes, run with it @MEZk . Thanks.

MEZk commented:
https://github.com/checkstyle/checkstyle/pull/3207
